src/program/memory.rs

```diff
@@ -90,6 +90,22 @@ impl Instruction {
     /// Return all memory accesses by the instruction - in expressions, captures, and memory manipulation
     pub fn get_memory_accesses(&self) -> MemoryAccesses {
         match self {
+            Instruction::BinaryLogic(BinaryLogic { operands, .. }) => {
+                let mut memories = HashSet::new();
+                memories.insert(operands.0.name.clone());
+                if let LogicalOperand::MemoryReference(mem) = &operands.1 {
+                    memories.insert(mem.name.clone());
+                }
+
+                MemoryAccesses {
```

Comment:
Don't the binary commands both read from and write to the first operand?

This wouldn't cause issues in the current system because reads will block all captures and writes, and there couldn't be a concurrent write happening, but it's worth keeping things correct in case that changes.

Reply:
Updated. I was a little murky on this section, and forgot to call-out a request for assist here. Would be great to also cover this in 1x1.